### PR TITLE
drivers/pipe: Don't use sched_[lock|unlock] to do protection

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since the sched lock can't work in SMP context

## Impact
SMP

## Testing
adb shell

### CODE REVIEW

**Comment by acassis** [comment]
@xiaoxiang781216 please fix the commit text:
Don't use shced_[lock|unlock] to do protection -> Don't use sched_[lock|unlock] to do protection

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 please fix the commit text: Don't use shced_[lock|unlock] to do protection -> Don't use sched_[lock|unlock] to do protection

@acassis, Done.

**Comment by xiaoxiang781216** [comment]
@acassis

File: drivers/pipes/pipe_common.c [hunk 1]
```diff
@@ -232,19 +232,18 @@ int pipecommon_open(FAR struct file *filep)
       dev->d_nreaders++;
     }
 
-  /* If opened for read-only, then wait for either (1) at least one writer
-   * on the pipe (policy == 0), or (2) until there is buffered data to be
-   * read (policy == 1).
-   */
+  while (!(filep->f_oflags & O_NONBLOCK) &&        /* Non-blocking */
```
Comment (pkarashchenko) [inline drivers/pipes/pipe_common.c:235]:
```suggestion
  while ((filep->f_oflags & O_NONBLOCK) == 0 &&        /* Non-blocking */
```

Comment (xiaoxiang781216) [reply drivers/pipes/pipe_common.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Review by acassis — APPROVED** [review]

**Pull request merged**